packages/eslint-plugin/docs/rules/prefer-at.md

```diff
@@ -0,0 +1,101 @@
+---
+description: 'Enforce the use of `array.at(-1)` instead of `array[array.length - 1]`.'
+---
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/prefer-at** for documentation.
+
+There are two ways to get the last item of the array:
+
+- `arr[arr.length - 1]`: getting an item by index calculated relative to the length of the array.
+- `arr.at(-1)`: getting an item by the `at` method. [Docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at)
+
+`arr.at(-1)` is a cleaner equivalent to `arr[arr.length - 1]`.
+
+This rule supports the following types:
+
+- `string` or `String`
+- `Array`
+- `Int8Array`
+- `Uint8Array`
+- `Uint8ClampedArray`
+- `Int16Array`
+- `Uint16Array`
+- `Int32Array`
+- `Float32Array`
+- `Uint32Array`
+- `Float64Array`
+- `BigInt64Array`
+- `BigUint64Array`
```

Comment:
[Proofreading] Folks probably don't care about the exact full list of these integer arrays, just that they're all supported.

```suggestion
`arr.at(-1)` is a cleaner equivalent to `arr[arr.length - 1]`.

This rule suggests code use `arr.at(-1)`, as it's cleaner than `arr[arr.length - 1]`.
It supports built-in array and string types, including integer arrays such as `Int8Array`.
```